src/nifler/configcmd.nim

```diff
@@ -203,8 +203,14 @@ proc produceConfig*(infile, outfile: string) =
     b.addHeader()
     b.withTree "config":
       b.buildConfig conf
-    conf.outDir = AbsoluteDir getCurrentDir()
-    conf.outFile = RelativeFile(outfile)
+    if outfile.isAbsolute:
+      let p = outfile.splitPath
+      conf.outDir = AbsoluteDir(p.head)
+      conf.outFile = RelativeFile(p.tail)
+    else:
+      conf.outDir = AbsoluteDir getCurrentDir()
+      conf.outFile = RelativeFile(outfile)
+
```

Comment:
Why was this change required?

Reply:
`outfile` can be an absolute path.
From `genSuccessX(conf)` proc, `absOutFile` proc in `https://github.com/nim-lang/Nim/blob/devel/compiler/options.nim#L723` is called.
`absOutFile` proc do `result = conf.outDir / conf.outFile`.
So `conf.outFile` must be relative.